airflow/kubernetes/pod_launcher.py

```diff
@@ -98,6 +98,12 @@ def delete_pod(self, pod: V1Pod):
             if e.status != 404:
                 raise
 
+    @tenacity.retry(
+        stop=tenacity.stop_after_attempt(3),
+        wait=tenacity.wait_random_exponential(),
+        reraise=True,
+        retry=tenacity.retry_if_exception_type(ApiException),
```

Comment:
What does ApiException cover? Would this also be the same error if you send invalid payload?

Reply:
I believe so, I think it would also cover 5xx errors due to invalid credentials in which case there definitely isn't any point to retrying. 

With this in mind, should we scope this retry down to only cover 409 errors?

Reply:
I think so, yeah probably.

Reply:
@ashb @SamWheating this might be a good reason to move the pod_launching code into the cncf.kubernetes package. I don't think K8sPodOperators should require dependencies on Airflow for these kinds of fixes.

Reply:
Ok, I just pushed a commit to only retry on 409 ApiExceptions, let me know what you think. 

Re: the code structure - it looks like the pod launching code is also used by the KubernetesExecutor, so if you wanted to move the pod_launcher to the CNCF provider package you would likely need a copy within the Airflow package as well?

Reply:
I think there are functions that are only used by the k8spodoperator/only used by the k8sexecutor, so wouldn't be TOO bad.

Reply:
Yeah, I just looked a little more and it appears that you're right. Its probably not too big of a change then to move a subset of the pod_launcher file into the cncf provider package and update some imports and such, and I'd definitely be interested in helping with that / taking that on. Would you like to write up an issue for that, or shall I?

Regarding the issue in question, would you be OK with reviewing/merging this PR in the meantime? A larger refactor might take a while to get properly reviewed and this issue is causing a lot of failures on our end.

Reply:
@SamWheating I've created an issue here https://github.com/apache/airflow/issues/15164. Please comment on it and I will assign it to you :).

I'd say you should attack this quickly, as without this fix we won't be able to release this fix until the next Airflow release  (and it will require an Airflow upgrade). 

That said, I'm glad to make this PR a high priority, so once it's ready I can be fast with PR reviews to get it through sooner than later.